src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/UpdateSentenceDateService.kt

```diff
@@ -130,106 +111,50 @@ class UpdateSentenceDateService(
-            dateFormat,
-          )
-        }" to sentenceChanges.prrdChanged,
-        "HDC actual date has changed to ${
-          if (updatedLicenceEntity is HdcLicence) {
-            updatedLicenceEntity.homeDetentionCurfewActualDate?.format(
-              dateFormat,
-            )
-          } else {
-            null
-          }
-        }" to sentenceChanges.hdcadChanged,
-        "HDC end date has changed to ${
-          if (updatedLicenceEntity is HdcLicence) {
-            updatedLicenceEntity.homeDetentionCurfewEndDate?.format(
-              dateFormat,
-            )
-          } else {
-            null
-          }
-        }" to sentenceChanges.hdcEndDateChanged,
-      ),
+      dateChanges.filter { it.changed && it.type.notifyOnChange }.map { it.toDescription() },
     )
   }
 
-  private fun logUpdate(
-    licenceId: Long,
-    licenceEntity: Licence?,
-    licenceStartDate: LocalDate?,
-    sentenceDates: SentenceDates,
-  ) {
+  private fun logUpdate(licenceId: Long, isMaterial: Boolean, dateChanges: List<DateChange>) {
     log.info(
       buildString {
         append("Licence dates - ID $licenceId ")
-        append("CRD ${licenceEntity?.conditionalReleaseDate} ")
-        append("ARD ${licenceEntity?.actualReleaseDate} ")
-        append("SSD ${licenceEntity?.sentenceStartDate} ")
-        append("SED ${licenceEntity?.sentenceEndDate} ")
-        append("LSD ${licenceEntity?.licenceStartDate} ")
-        append("LED ${licenceEntity?.licenceExpiryDate} ")
-        append("TUSSD ${licenceEntity?.topupSupervisionStartDate} ")
-        append("TUSED ${licenceEntity?.topupSupervisionExpiryDate} ")
-        append("PRRD ${licenceEntity?.postRecallReleaseDate}")
-        if (licenceEntity is HdcLicence) {
-          append("HDCAD ${licenceEntity.homeDetentionCurfewActualDate} ")
-        }
+        dateChanges.forEach { append("${it.type.name} ${it.oldDate} ") }
       },
     )
-
     log.info(
       buildString {
         append("Event dates - ID $licenceId ")
-        append("CRD ${sentenceDates.conditionalReleaseDate} ")
-        append("ARD ${sentenceDates.actualReleaseDate} ")
-        append("SSD ${sentenceDates.sentenceStartDate} ")
-        append("SED ${sentenceDates.sentenceEndDate} ")
-        append("LSD $licenceStartDate ")
-        append("LED ${sentenceDates.licenceExpiryDate} ")
-        append("TUSSD ${sentenceDates.topupSupervisionStartDate} ")
-        append("TUSED ${sentenceDates.topupSupervisionExpiryDate} ")
-        append("PRRD ${sentenceDates.postRecallReleaseDate}")
-        if (licenceEntity is HdcLicence) {
-          append("HDCAD ${sentenceDates.homeDetentionCurfewActualDate} ")
-        }
+        dateChanges.forEach { append("${it.type.name} ${it.newDate} ") }
+      },
+    )
+    log.info(
+      buildString {
+        append("Date change flags: ")
+        dateChanges.forEach { append("${it.type.name} ${it.changed} ") }
+        append("isMaterial $isMaterial")
```

Comment:
Might simplify all this logging and merge into a single log statement in a follow up